testing/pom.xml

```diff
@@ -23,5 +23,15 @@
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.cucumber</groupId>
+            <artifactId>cucumber-java</artifactId>
+            <version>6.10.4</version>
```

Comment:
This should probably be test scope, no?

Reply:
probably

Reply:
I've thought a bit more about this. I think there are two options here - `compile` scope or `provided` scope. (See [maven docs](https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#dependency-scope) for details).

The `junit` dependency is already `compile` scope, so using the same scope for `cucumber-java` and `cucumber-junit` would be consistent with that.

Alternatively, making it `provided` scope would indicate that dependent modules would have to explicitly add the dependencies themselves. I think that's ok too.

I'll defer to you to decide what's most appropriate.

Reply:
If someone is already using this framework (our Java-testing) without cucumber and I merge this PR. Regardless of the provided or not, I believe the build will fail for not finding the dependency in the classpath, right? 

I need time to sync internally but if that's the case we may need to split this into a separate submodule.

Reply:
> I believe the build will fail for not finding the dependency in the classpath, right?

No it won't.

In Java, classes are only loaded when they are referenced by another class. Classes are not loaded by mere presence in a jar file.

No other classes in this library reference `io.split.client.testing.cucumber.CucumberSplit`, so it won't be loaded.